controller/refugee_controller.py

```diff
@@ -21,31 +22,41 @@ def create_refugee(firstname: str,
                    starting_date: date,
                    medical_condition_type: Refugee.MedicalCondition) -> Refugee:
     """
-    Yingbo, Michelle
-    A function used by admin and volunteer.
+    Function to create and save refugee data.
     """
-    pass
+    try:
+        new_refugee = Refugee(firstname=firstname,
+                              lastname=lastname,
+                              num_of_family_member=num_of_family_member,
+                              starting_date=starting_date,
+                              medical_condition_type=medical_condition_type)
+        camp.refugees.add(new_refugee)
+        return new_refugee
+    except (Refugee.InvalidNameException,
+            Refugee.InvalidNumOfFamilyMemberException,
+            Refugee.InvalidStartingDateException,
+            Refugee.InvalidCampException) as e:
+        raise ControllerError(str(e))
 
 
 def find_refugee(refugee_id: int) -> Refugee:
     """
-    Yingbo, Michelle
-
+    Function to find if refugee exists and returns the refugee class profile. Raise error if no such refugee exist.
     """
-    pass
+    try:
+        for plan in Plan.all():
+            for camp in plan.camps:
+                refugee = camp.refugees.get(refugee_id)
+                if refugee is not None:
+                    return refugee
+                else:
+                    raise ControllerError(f"Invalid refugee refugee_id: {refugee_id}. The refugee is not found.")
```

Comment:
This looks wrong. You are raising the error when you cannot find the refugee at the first camp in the first plan. The refugee can be in other camps or plans. 
What you should do is raise error after checking all plans and camps

Reply:
Updated, please have a look if the current error is okay!